### ISSUE TITLE
How to change data in measurements and send it?

### ISSUE THREAD

**Issue by lu003** [Azure/iotc-device-bridge#68]
Hi! Great tutorial and this is not an issue with the bridge/tutorial at all, but am unsure where else to post this question besides stack overflow. I'm very new to Node.js and I'm having trouble finding a way to change the data in measurements before sending it. I'm working with Particle and my boss wanted the Particle.publish event value data to be a string that has all the actual data values in, so now when the webhook and function sends it to IoT Central in measurements there is only one piece of data (which is the string). Currently, I'm trying to change it in the Azure function, but am finding it difficult to decipher. I looked into context and was able to split my data, but am unsure how to correctly assign it in measurements and send it. Is there a way I can manipulate the data in measurements before sending it to IoT Central, such that I can split the data in the string and have a new measurements JSON sent with the values I want to display? 

This is my current Particle JSON:
```
{
  "device": {
    "deviceId": "{{{PARTICLE_DEVICE_ID}}}"
  },
  "measurements": {
    "allData": "{{{PARTICLE_EVENT_VALUE}}}"
  }
}
```

**Comment by peolivei2** [comment]
Measurements are passed to the `handleMessage` function in the main file: https://github.com/Azure/iotc-device-bridge/blob/master/IoTCIntegration/index.js#L21

At this point, you need to replace `req.body.measurements` by your decoded measurements (an object that includes one key/value per measurement). This depends on which format your measurements are being sent.

For instance, from your example, let's suppose that `allData` had the stringified JSON representing all measurements. You'd have to replace `req.body.measurements` by `JSON.parse(req.body.measurements.allData)`. Now, if `allData` was a string of comma-separated values, then the decoding would be different. Again, all depends on how the `allData` string is formatted. You have to make sure you parse it before passing to `handleMessage`.

**Comment by lu003** [comment]
Hi @peolivei2,
Thank you for such a quick response! I knew the `handleMessage` was what sent it I just didn't know how it worked and how `req.body.measurements` was accessed because I tried to access it before with `req.body.measurements.allData` and it was undefined. Fortunately, I was able to just fill a new JSON with the data in context and passed it in like you said. Thank you again for all your help!
